code/game/atoms.dm

```diff
@@ -74,6 +74,9 @@
 	/// Radiation insulation types
 	var/rad_insulation = RAD_NO_INSULATION
 
+	/// The icon state intended to be used for the acid component. Used to override the default acid overlay icon state.
+	var/custom_acid_overlay = null
```

Comment:
Can't this be stored on the component?

Reply:
there's some issues with doing that because an object doesn't apply acid itself, it gets acid applied to it, the only other way of doing this is having a list that keeps the icons listed in an assoc list with type as key but honestly thats not great either.

Reply:
Yea, that'd offer no gains.
I suppose that short of an acidable element this is fine.